### PR TITLE
Make Route Guide client and server accept channel and server builders

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

File: examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java [hunk 1]
```diff
@@ -60,9 +60,12 @@
...
   /** Construct client for accessing RoutGuide server at {@code host:port}. */
   public RouteGuideClient(String host, int port) {
-    channel = ManagedChannelBuilder.forAddress(host, port)
-        .usePlaintext(true)
-        .build();
+    this(ManagedChannelBuilder.forAddress(host, port).usePlaintext(true));
+  }
+
+  /** Construct client for accessing RoutGuide server using the existing channel. */
+  public RouteGuideClient(ManagedChannelBuilder<?> channelBuilder) {
```
Comment (ejona86) [inline examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java]: Why pass in the builder vs the `ManagedChannel` itself?

File: examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java [hunk 2]
```diff
@@ -60,9 +60,12 @@
 
   /** Construct client for accessing RoutGuide server at {@code host:port}. */
   public RouteGuideClient(String host, int port) {
-    channel = ManagedChannelBuilder.forAddress(host, port)
-        .usePlaintext(true)
-        .build();
+    this(ManagedChannelBuilder.forAddress(host, port).usePlaintext(true));
+  }
+
+  /** Construct client for accessing RoutGuide server using the existing channel. */
```
Comment (ejona86) [inline examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java]: nit: s/Rout/Route/

File: examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java (hunk 1 above)
Comment (carl-mastrangelo) [inline examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java]: Mostly just to match the server construction.

**Comment by ejona86** [comment]
@carl-mastrangelo LGTM, with one nit

File: examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java (hunk 2 above)
Comment (carl-mastrangelo) [inline examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java]: Whoops, that'll teach me to copy-paste.

Done.

**Pull request merged**